sync-core/src/main/java/com/cloudant/mazha/CouchClient.java

```diff
@@ -50,61 +47,45 @@
     public static final String COUCH_ERROR_CONFLICT = "conflict";
 
     private final HttpRequests httpClient;
-    private final JSONHelper json;
-    private final String defaultDb;
+    private final JSONHelper jsonHelper;
     private CouchURIHelper uriHelper;
 
-    public CouchClient(CouchConfig config, String dbName) {
+    public CouchClient(CouchConfig config) {
         this.httpClient = new HttpRequests(this.getHttpConnectionParams(config),
-                config.getUsername(),
-                config.getPassword(),
+                config.getRootUri().getUserInfo(),
                 config.getCustomHeaders());
-        this.defaultDb = dbName;
-        this.json = new JSONHelper();
-        this.uriHelper = new CouchURIHelper(
-                config.getProtocol(),
-                config.getHost(),
-                config.getPort()
-        );
+        this.jsonHelper = new JSONHelper();
+        this.uriHelper = new CouchURIHelper(config.getRootUri());
     }
 
+    // TODO rename
     public URI getDefaultDBUri() {
-        return this.uriHelper.dbUri(this.defaultDb);
-    }
-
-    public String getDefaultDb() {
-        return this.defaultDb;
+        return this.uriHelper.dbUri();
     }
 
     HttpRequests getHttpClient() {
         return httpClient;
     }
 
-    protected JSONHelper getJson() {
-        return json;
-    }
-
-    public List<String> allDbs() {
-        URI allDBUri = uriHelper.allDbsUri();
-        return getList(allDBUri);
+    protected JSONHelper getJsonHelper() {
+        return jsonHelper;
     }
```

Comment:
You use `this.jsonHelper` in `revsDiff` rather than the accessor. As jsonHelper is final, you could perhaps just make the ivar protected rather than having the accessor.